Modules/_functoolsmodule.c

```diff
@@ -1175,7 +1176,11 @@ uncached_lru_cache_wrapper(lru_cache_object *self, PyObject *args, PyObject *kwd
 {
     PyObject *result;
 
+#ifdef Py_GIL_DISABLED
+    _Py_atomic_add_ssize(&self->misses, 1);
+#else
     self->misses++;
+#endif
```

Comment:
Could we define FT_ATOMIC_ADD_SSIZE or FT_ATOMIC_INCR_SSIZE, even if just in this file?
```c
#ifdef Py_GIL_DISABLED
#define FT_ATOMIC_ADD_SSIZE(value, new_value) \
    _Py_atomic_add_ssize(&value, new_value)
#else
#define FT_ATOMIC_ADD_SSIZE(value, new_value) value += new_value
#endif
```
```c
#ifdef Py_GIL_DISABLED
#define FT_ATOMIC_INCR_SSIZE(value) \
    _Py_atomic_add_ssize(&value, 1)
#else
#define FT_ATOMIC_INCR_SSIZE(value) value++
#endif
```

Reply:
I almost did but didn't want to decide between returning the previous value "_Py_atomic_add_ssize()" vs the new value "value += new_value". If you want to make the call I can add it.

Reply:
Good point. Then FT_ATOMIC_INCR_SSIZE should not create such dilemma.